Tests/ASTextNodeTests.m

```diff
@@ -255,6 +256,32 @@ - (void)testThatTheExperimentWorksCorrectly
   XCTAssertEqualObjects(sc2.superclass, [ASTextNodeSubclass class]);
 }
 
+- (void)testTextNodeSwitchWorksInMultiThreadEnvironment
+{
+  ASConfiguration *config = [ASConfiguration new];
+  config.experimentalFeatures = ASExperimentalTextNode;
+  [ASConfigurationManager test_resetWithConfiguration:config];
+  XCTestExpectation *exp = [self expectationWithDescription:@"wait for full bucket"];
+
+  dispatch_queue_t queue = dispatch_queue_create("com.texture.AsyncDisplayKit.ASTextNodeTestsQueue", DISPATCH_QUEUE_CONCURRENT);
+  dispatch_group_t g = dispatch_group_create();
+  for (int i = 0; i < 20; i++) {
+    dispatch_group_async(g, queue, ^{
+      ASTextNode *textNode = [[ASTextNodeSecondSubclass alloc] init];
+      XCTAssert([textNode isKindOfClass:[ASTextNode2 class]]);
+      @synchronized(self.textNodeBucket) {
```

Comment:
Seems like the dispatch group ended up not getting used, so let's remove it.

Reply:
Actually the test fails very rarely may be 1/30 when I tested locally for pre-fix code.
Also 
https://developer.apple.com/documentation/objectivec/1441536-class_setsuperclass?language=objc suggest we not use method. 
Any way to avoid this at all and achieve same idea without this runtime change of super class?

Reply:
Since we change the superclass before allocating any instances, I expect it's relatively safe for this limited usage. If the test is flaky then I think we should skip this diff – I doubt this issue will return now that the fix has landed.